cocotb/triggers.py

```diff
@@ -748,15 +775,15 @@ def with_timeout(triggers, timeout_time, timeout_unit=None):
             Any expression that could be right of a :keyword:`yield`
             (or :keyword:`await` in Python 3) expression in cocotb.
         timeout_time (int):
```

Comment:
Is this actually a ``numbers.Number`` too? At least floats are handled here, right? Same goes for ``Timer``s ``time_ps`` argument.

Reply:
Turns out I documented the other one wrongly - strictly this is `numbers.Real or decimal.Decimal`, since complex numbers are forbidden, and `Decimal` isn't a number at all!

Reply:
I now changed this and ``get_sim_steps`` to ``numbers.Real or decimal.Decimal`` but left ``Timer(time_ps)`` without type description since the examples are doing a good job for that.